numpy/core/src/umath/simd.inc.src

```diff
@@ -1671,6 +1700,95 @@ static NPY_INLINE NPY_GCC_OPT_3 NPY_GCC_TARGET_@ISA@ @vtype@d
 #endif
 /**end repeat**/
 
+/**begin repeat
+ * #type = npy_float, npy_double#
+ * #TYPE = FLOAT, DOUBLE#
+ * #num_lanes = 16, 8#
+ * #vsuffix = ps, pd#
+ * #mask = __mmask16, __mmask8#
+ * #vtype = __m512, __m512d#
+ * #scale = 4, 8#
+ * #vindextype = __m512i, __m256i#
+ * #vindexsize = 512, 256#
+ * #vindexload = _mm512_loadu_si512, _mm256_loadu_si256#
+ */
+
+/**begin repeat1
+ *  #func = maximum, minimum#
+ *  #vectorf = max, min#
+ */
+
+#if defined HAVE_ATTRIBUTE_TARGET_AVX512F_WITH_INTRINSICS && defined NPY_HAVE_SSE2_INTRINSICS
+static NPY_INLINE NPY_GCC_TARGET_AVX512F void
+AVX512F_@func@_@TYPE@(char **args, npy_intp *dimensions, npy_intp *steps)
+{
+    const npy_intp stride_ip1 = steps[0]/sizeof(@type@);
+    const npy_intp stride_ip2 = steps[1]/sizeof(@type@);
+    const npy_intp stride_op = steps[2]/sizeof(@type@);
+    const npy_intp array_size = dimensions[0];
+    npy_intp num_remaining_elements = array_size;
+    @type@* ip1 = (@type@*) args[0];
+    @type@* ip2 = (@type@*) args[1];
+    @type@* op  = (@type@*) args[2];
+
+    @mask@ load_mask = avx512_get_full_load_mask_@vsuffix@();
+
+    npy_int index_ip1[@num_lanes@], index_ip2[@num_lanes@], index_op[@num_lanes@];
```

Comment:
This needs to be `npy_intp`

Reply:
It works fine for large array sizes but it will fail for large strides. The maximum value of `index_ip1`, `index_ip2 `and `index_op `is `16*stride `, So it will overflow when `stride > 2^28 = 268435456 `.

Reply:
Any reason not to use
```suggestion
    /* note: while generally indices are npy_intp, we ensure that our maximum index
     * will fit in an int32 as a precondition for this function via IS_OUTPUT_BLOCKABLE_UNARY
     */
    npy_int32 index_ip1[@num_lanes@], index_ip2[@num_lanes@], index_op[@num_lanes@];
```